src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -777,4 +777,16 @@ public long countSubmissionsByCourseIdSubmitted(Long courseId) {
         log.debug("countSubmissionsByCourseIdSubmitted took " + (System.currentTimeMillis() - start) + "ms");
         return count;
     }
+
+    /**
+     * Sets the transient attribute "testRepositoryUrl" if the exercises is a programming exercise
+     *
+     * @param exercise the exercise for which to set the url
+     */
+    public void setTestRepositoryUrlForProgrammingExercise(Exercise exercise) {
```

Comment:
please rename the method, it does not reflect what is done here.

Reply:
i renamed the method to checksAndSetsIfProgrammingExerciseIsLocalSimulation